public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/unwrap.ts

```diff
@@ -0,0 +1,6 @@
+export function unwrap<T>(value: T | null | undefined): T {
```

Comment:
Is it an assertion just for the sake of refactoring or do we want to keep it?

Reply:
i want to keep it :-) 

it is there to make typescript happy, otherwise you get problems where typescript thinks something can be nullish, when you in fact know that it cannot.

for example, if you have a `grafana-ui/Select`, it works with `SelectableValue` objects ( https://github.com/grafana/grafana/blob/45c763a76bbf94c88be6ba13ffbe21381e266666/packages/grafana-data/src/types/select.ts#L6 ) , so you get one in `onChange`,  but the `.value` attribute can be `undefined` too. i know i never gave it one with `undefined` so i know i should never get that value. so if the value is undefined, programming error happened, and i just want things to crash with an exception :-) that's what `unwrap` does.

(i know there is the `!` operator ( https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator ) , but i don't like that it never checks it, so you will not be notified when your not-nullish assumption fails.

for example, somewhere in this issue, you mention an error-message when something goes wrong with "Uncaught Error: value must not be nullish" . that is produced by this unwrap-function :), and yes, i should fix that place.

(the name might not be the best, it's inspired by a similar thing in rust ( https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.unwrap )

Reply:
(i'm not sure what the etiquette is for resolving such discussions, for now i will resolve it, if you disagree feel free to reopen it and we can continue the discussion)